sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalog.scala

```diff
@@ -0,0 +1,167 @@
+  }
+
+  override def listTables(namespace: Array[String]): Array[Identifier] = {
+    checkNamespace(namespace)
+    withConnection { conn =>
+      val md = conn.getMetaData
+      val schemaPattern = if (namespace.length == 1) namespace.head else null
+      val rs = md.getTables(null, schemaPattern, "%", Array("TABLE"));
+      val tables = ArrayBuffer.empty[Identifier]
+      while (rs.next()) {
+        tables += Identifier.of(namespace, rs.getString("TABLE_NAME"))
+      }
+      tables.toArray
+    }
+  }
+
+  override def tableExists(ident: Identifier): Boolean = {
+    val ns = ident.namespace()
+    checkNamespace(ns)
+    withConnection { conn =>
+      val md = conn.getMetaData
+      val schemaPattern = if (ns.length == 1) ns.head else null
+      val rs = md.getTables(null, schemaPattern, ident.name(), Array("TABLE"));
+      rs.next()
+    }
+  }
+
+  override def renameTable(oldIdent: Identifier, newIdent: Identifier): Unit = {
+    // TODO: support this by adding more APIs to `JdbcDialect` which can generate the SQL statement
+    //       to rename a table.
+    throw new UnsupportedOperationException("rename table")
+  }
+
+  override def alterTable(ident: Identifier, changes: TableChange*): Table = {
+    // TODO: support this by adding more APIs to `JdbcDialect` which can generate the SQL statement
+    //       to alter a table.
+    throw new UnsupportedOperationException("alter table")
+  }
+
+  override def dropTable(ident: Identifier): Boolean = {
+    checkNamespace(ident.namespace())
+    try {
+      withConnection { conn =>
+        val statement = conn.createStatement
+        statement.setQueryTimeout(options.queryTimeout)
+        statement.executeUpdate(s"DROP TABLE ${getTableName(ident)}")
+        true
+      }
+    } catch {
+      case _: SQLException => false
+    }
+  }
+
+  override def loadTable(ident: Identifier): Table = {
+    checkNamespace(ident.namespace())
+    val optionsWithTableName = createOptionsWithTableName(ident)
+    try {
+      val schema = JDBCRDD.resolveTable(optionsWithTableName)
+      JDBCTable(ident, schema, optionsWithTableName)
+    } catch {
+      case _: SQLException => throw new NoSuchTableException(ident)
+    }
+  }
+
+  override def createTable(
+      ident: Identifier,
+      schema: StructType,
+      partitions: Array[Transform],
+      properties: util.Map[String, String]): Table = {
+    checkNamespace(ident.namespace())
+    if (partitions.nonEmpty) {
+      throw new UnsupportedOperationException("Cannot create JDBC table with partition")
+    }
+    // TODO: we can support this, but we need to add an API to `JdbcDialect` to generate the SQL
+    //       statement to specify table options.
+    if (!properties.isEmpty) {
+      logWarning("Cannot create JDBC table with properties, these properties will be " +
+        "ignored: " + properties.asScala.map { case (k, v) => s"$k=$v" }.mkString("[", ", ", "]"))
```

Comment:
For security, shall we redaction the value? Or, it will be enough to print only `key` list?
cc @gatorsmile

Reply:
@dongjoon-hyun good point, +1